map.go

```diff
@@ -1614,6 +1637,85 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 	return false
 }
 
+func (mi *MapIterator) nextDrain(keyOut, valueOut interface{}) bool {
+	// Check for maps without a key (e.g., Queue/Stack). In this case, when there is no
+	// more data, ErrKeyNotExist arise, but we gently stop the retrieval with no error.
+	if mi.target.keySize == 0 {
+		mi.err = mi.target.LookupAndDelete(keyOut, valueOut)
```

Comment:
This allows a user to provide a non-nil key even when we know that there probably shouldn't be one. I'm on the fence whether this is a good idea. Maybe we should validate that `key == nil` instead?